### ISSUE TITLE
Incorrect parsing when string contains }} (two closing braces)

### ISSUE THREAD

**Issue by glen-84** [vuejs/vue-eslint-parser#94]
https://astexplorer.net/#/gist/50f0fc500a626ee33ec2ea9205f4e598/8be358100a557256023ef600d4aa7e32486ea369

Two closing braces inside a string should not be seen as the end of the expression container.

This will happen frequently when using something like ICU message formatting, as seen in the example at the URL above.

**Comment by ota-meshi** [comment]
I think your template is incorrect as SFC.

- [Vue 2 Template Explorer](https://template-explorer.vuejs.org/#%20%20%3Cp%3E%7B%7B%20%24fm('%7BserverCount%2C%20plural%2C%20one%20%7BServer%7D%20other%20%7BServers%7D%7D')%20%7D%7D%3C%2Fp%3E)
- [Vue 3 Template Explorer](https://vue-next-template-explorer.netlify.app/#%7B%22src%22%3A%22%20%20%3Cp%3E%7B%7B%20%24fm('%7BserverCount%2C%20plural%2C%20one%20%7BServer%7D%20other%20%7BServers%7D%7D')%20%7D%7D%3C%2Fp%3E%22%2C%22ssr%22%3Afalse%2C%22options%22%3A%7B%22mode%22%3A%22module%22%2C%22prefixIdentifiers%22%3Afalse%2C%22optimizeImports%22%3Afalse%2C%22hoistStatic%22%3Atrue%2C%22cacheHandlers%22%3Afalse%2C%22scopeId%22%3Anull%2C%22inline%22%3Afalse%2C%22ssrCssVars%22%3A%22%7B%20color%20%7D%22%2C%22bindingMetadata%22%3A%7B%22TestComponent%22%3A%22setup%22%2C%22foo%22%3A%22setup%22%2C%22bar%22%3A%22props%22%7D%2C%22optimizeBindings%22%3Afalse%7D%7D)

So I think this parser is working fine.

**Comment by longlho** [comment]
I'm not sure I understand how it's incorrect?
```vue
<template>
  <p>{{ $fm('}}') }}</p>
</template>
```

This looks like it should be parsable.

**Comment by ota-meshi** [comment]
Your template cannot be compiled with Vue's compiler.
What do you see and think this template can be parsed?

![image](https://user-images.githubusercontent.com/16508807/103617556-ce8a4900-4f71-11eb-9cc9-f30a22503b5c.png)


https://template-explorer.vuejs.org/#%3Cp%3E%7B%7B%20%24doSomething('%7D%7D')%20%7D%7D%3C%2Fp%3E

**Comment by LinusBorg** [comment]
`'}}'` is interpreted by the parser as the end of an interpolation, but it isn't. This looks to me like a bug or improvement opportunity for the Vue parser.

**Comment by glen-84** [comment]
So this is an issue with `@vue/compiler-*`?

Does GH allow you to transfer this issue there, or should I open a new issue at `vuejs/vue-next`?

**Comment by LinusBorg** [comment]
I can't transfer it, maybe @ota-meshi can?

**Comment by ota-meshi** [comment]
@LinusBorg Thank you for following me.
I can't even transfer this issue...

**Comment by LinusBorg** [comment]
@glen-84 Then please open a new issue in the vue repo

**Comment by glen-84** [comment]
I will do, thanks.

**Comment by LinusBorg** [comment]
FWIW I guess you are aware that a workaround would be `'}' + '}'`

**Comment by glen-84** [comment]
Ya, @longlho put [another workaround](https://formatjs.io/docs/vue-intl#using-icu-in-vue-sfc) in the Format.JS documentation, but I wanted to find out why it was necessary. 🙂

I've opened https://github.com/vuejs/vue-next/issues/2944.

Thanks.
